src/mpid/ch4/generic/gpu/mpidig_gpu_utils.h

```diff
@@ -0,0 +1,132 @@
+/*
+ *  Copyright (C) by Argonne National Laboratory.
+ *      See COPYRIGHT in top-level directory.
+ */
+#ifndef MPIDIG_GPU_UTIL_H_INCLUDED
+#define MPIDIG_GPU_UTIL_H_INCLUDED
+
+#include "ch4_impl.h"
+
+/* MPIDIG_gpu_alloc_stage_buf - allocate contiguous stage buffer and return pointer */
+MPL_STATIC_INLINE_PREFIX void *MPIDIG_gpu_alloc_stage_buf(MPI_Aint count, MPI_Datatype datatype)
+{
+    char *tmp_buf = NULL;
+    size_t size;
+    MPIDI_Datatype_check_size(datatype, count, size);
+    MPL_gpu_malloc_host((void **) &tmp_buf, size);
+
+    return tmp_buf;
+}
```

Comment:
I am not sure about the utility of this function anymore. In the AM code I am working on there are cases in which I need to allocate space not only for the data itself but also for the AM headers. The only advantage of this function is that it calculates the buffer size internally. But in many parts of the code we already do this anyways. Maybe `MPL_gpu_malloc/free_host` is enough.

Reply:
True. The current interface only works with pure data allocation, not header.
In OFI or AM, we do already have `data_ptr` and `data_sz` calculated, before appropriate place  of calling this function. 
I'm fine with calling `MPL_gpu_malloc/free_host` directly.

Reply:
@pavanbalaji What do you think?